totalRP3_Extended/aura/aura.lua

```diff
@@ -0,0 +1,1009 @@
+local getClass = TRP3_API.extended.getClass;
+
+local TRP3_AuraFrame, TRP3_AuraFrameCollapseAndExpandButton, TRP3_InspectionFrame = TRP3_AuraFrame, TRP3_AuraFrameCollapseAndExpandButton, TRP3_InspectionFrame;
+
+local getRootClassID = TRP3_API.extended.getRootClassID;
+
+local EMPTY = Globals.empty;
+
+--[[
+	maximum number of iterations within one auraCore:Update() batch
+	this is to avoid calling Update() recursively and also to repaint the UI only once
+]]--
+local MAX_AURA_UPDATE_CYCLES = 10;
+
+--[[
+	minimum time to pass until the next automated call to auraCore:Update()
+	Update calls via event do not adhere to this minimum, however Update will ensure
+	that there is at maximum one active C_Timer
+]]--
+local MIN_AURA_UPDATE_INTERVAL = 0.1;
+
+local TOOLTIP_REFRESH_INTERVAL = 0.2;
+
+--[[
+	Update frequency for dynamic auras. Since variables can change at any time, let's just update
+	auras every DYN_AURA_UPDATE_INTERVAL seconds, if they happen to display variables
+]]--
+local DYN_AURA_UPDATE_INTERVAL = 1;
+
+local INSPECT_AURA_OFFSET = 10;
+local INSPECT_AURA_HEIGHT = 36;
+local INSPECT_AURA_COL_WIDTH = 36;
+local INSPECT_AURAS_PER_COL = 8;
+
+local AURA_HEIGHT = 48;
+local AURA_WIDTH = 36;
+local AURA_MARGIN = 16;
+local AURA_MARGIN_TOP = 24;
+local AURA_ROW_LENGTH = 8;
+
+--[[ DATA MAP
+
+	            auraCore.activeAuras[i]                        TRP3_API.profile.getPlayerCurrentProfile().auras[j]
+	
+	                     |                                                              |
+	                    \|/                                                            \|/
+	
+	+-----------------------------------------------+          +------------------------------------------------+
+	| t persistent  pointer to stored part ---------|--------> | s id            full class id                  |
+	| t class       cached class data (pointer)     |          | t vars          aura variables                 |
+	| s duration    remaining duration string       |          | n expiry        expiry timestamp               |
+	| s overlay     overlay string                  |          | n dormantSince  timestamp of last deactivation |
+	| b initialize  flag to run the onInit script   |          | n lastTick      timestamp of the last onTick   |
+	| b expired     flags the aura as expired       |          | b invalid       invalidated flag (aura should  |
+	| b cancel      flag to run the onCancel script |          |                 be removed on next Update)     |
+	| b cancelled   flags the aura as cancelled     |          +------------------------------------------------+
+	| b hasDynamicDescription   whether or not the  |
+	|               description text has tags       |
+	| b hasDynamicOverlay       whether or not the  |
+	|               overlay text has tags           |
+	+-----------------------------------------------+
+
+]]--
+
+local CUSTOM_EVENTS = TRP3_API.extended.CUSTOM_EVENTS;
+
+local auraCore;
+auraCore = {
```

Comment:
General style comment - as far as I'm aware nothing else in the addon creates _entire modules_ whose methods are defined as methods directly in the table constructor literal. It looks out of place and consumes an indentation level a bit pointlessly.

Would recommend just blanket cleaning this up to the more standard approach of:

```lua
local AuraCore = {};

function AuraCore:Now()
    return self.baseTime + GetTime();
end

function AuraCore:Initialize()
    ...
end
```